WordPress/Classes/ViewRelated/Me/All Domains/View Models/AllDomainsListViewModel.swift

```diff
@@ -40,6 +48,43 @@ class AllDomainsListViewModel {
         try? WPAccount.lookupDefaultWordPressComAccount(in: contextManager.mainContext)
     }
 
+    // MARK: - Resolving State
+
+    /// Determines the state of the view based on the domains and an optional search query.
+    ///
+    /// - Parameters:
+    ///   - domains: An array of domain objects to be filtered.
+    ///   - searchQuery: An optional search query to filter the domains. Pass `nil` or an empty string to skip filtering.
+    /// - Returns: The `.normal` or `.empty` state.
+    private func state(from domains: [Domain], searchQuery: String?) -> State {
+        if domains.isEmpty {
+            return .empty(emptyStateViewModel())
+        }
+
+        var domains = domains
+
+        if let searchQuery, !searchQuery.trimmingCharacters(in: .whitespaces).isEmpty {
+            domains = domains.filter { domain in
+                return domain.domain.localizedStandardContains(searchQuery)
+                || domain.siteSlug.localizedStandardContains(searchQuery)
+                || domain.blogName.localizedStandardContains(searchQuery)
+                || (domain.status?.value.localizedStandardContains(searchQuery) ?? false)
```

Comment:
[Suggestion] I suggest moving this to a function under `Domain`, and here just use something like `domain.matches(query: searchQuery)`

Reply:
Resolved in 02e7c3104e62ef04f4545788b5ada566cdc21a09